main/lsp/LSPPreprocessor.cc

```diff
@@ -345,112 +243,68 @@ void LSPPreprocessor::preprocessAndEnqueue(QueueState &state, unique_ptr<LSPMess
     if (config->isUriInWorkspace(uri)) {
         string localPath = config->remoteName2Local(uri);
         if (!config->isFileIgnored(localPath)) {
-            updates.updatedFiles.push_back(make_shared<core::File>(
-                move(localPath), move(openParams->textDocument->text), core::File::Type::Normal, v));
+            edit->updates.push_back(make_shared<core::File>(move(localPath), move(openParams->textDocument->text),
+                                                            core::File::Type::Normal, v));
         }
     }
+    return edit;
 }
 
-void LSPPreprocessor::canonicalizeEdits(u4 v, unique_ptr<DidCloseTextDocumentParams> closeParams,
-                                        LSPFileUpdates &updates) const {
-    updates.versionStart = v;
-    updates.versionEnd = v;
+unique_ptr<SorbetWorkspaceEditParams>
+LSPPreprocessor::canonicalizeEdits(u4 v, unique_ptr<DidCloseTextDocumentParams> closeParams) const {
+    auto edit = make_unique<SorbetWorkspaceEditParams>();
+    edit->epoch = v;
     string_view uri = closeParams->textDocument->uri;
     if (config->isUriInWorkspace(uri)) {
         string localPath = config->remoteName2Local(uri);
         if (!config->isFileIgnored(localPath)) {
             // Use contents of file on disk.
-            updates.updatedFiles.push_back(make_shared<core::File>(
-                move(localPath), readFile(localPath, *config->opts.fs), core::File::Type::Normal, v));
+            edit->updates.push_back(make_shared<core::File>(move(localPath), readFile(localPath, *config->opts.fs),
+                                                            core::File::Type::Normal, v));
         }
     }
+    return edit;
 }
 
-void LSPPreprocessor::canonicalizeEdits(u4 v, unique_ptr<WatchmanQueryResponse> queryResponse,
-                                        LSPFileUpdates &updates) const {
-    updates.versionStart = v;
-    updates.versionEnd = v;
+unique_ptr<SorbetWorkspaceEditParams>
+LSPPreprocessor::canonicalizeEdits(u4 v, unique_ptr<WatchmanQueryResponse> queryResponse) const {
+    auto edit = make_unique<SorbetWorkspaceEditParams>();
+    edit->epoch = v;
     for (auto file : queryResponse->files) {
         // Don't append rootPath if it is empty.
         string localPath = !config->rootPath.empty() ? absl::StrCat(config->rootPath, "/", file) : file;
         // Editor contents supercede file system updates.
         if (!config->isFileIgnored(localPath) && !openFiles.contains(localPath)) {
-            updates.updatedFiles.push_back(make_shared<core::File>(
-                move(localPath), readFile(localPath, *config->opts.fs), core::File::Type::Normal, v));
+            edit->updates.push_back(make_shared<core::File>(move(localPath), readFile(localPath, *config->opts.fs),
+                                                            core::File::Type::Normal, v));
         }
     }
-}
-
-void LSPPreprocessor::mergeEdits(LSPFileUpdates &to, LSPFileUpdates &from) {
```

Comment:
The preprocessor no longer deals with `LSPFileUpdates` objects, which contain a lot of stuff the preprocessor no longer cares about (like parsed ASTs). It only deals with LSP messages, and merging adjacent `SorbetWorkspaceEdit` messages.